File: app/src/main/java/com/hfad/avc/ui/list/all_congratulations_list/AllCongratulationsListPresenter.java
package com.hfad.avc.ui.list.all_congratulations_list;

import com.hfad.avc.dagger.ComponentManager;

import moxy.InjectViewState;
import moxy.MvpPresenter;

@InjectViewState
public class AllCongratulationsListPresenter extends MvpPresenter<IAllCongratulationsListViewModel> {

   public AllCongratulationsListPresenter() {
      ComponentManager.Companion.getInstance().appComponent.inject(this);
   }

   @Override
   public void attachView(IAllCongratulationsListViewModel view) {
      super.attachView(view);
      getViewState().setData();
   }
}

File: app/src/main/java/com/hfad/avc/ui/list/all_congratulations_list/AllCongratulationsListPresenter.kt
package com.hfad.avc.ui.list.all_congratulations_list

import com.hfad.avc.dagger.ComponentManager.Companion.instance
import moxy.InjectViewState
import moxy.MvpPresenter


@InjectViewState
class AllCongratulationsListPresenter :
   MvpPresenter<IAllCongratulationsListViewModel>() {
   init {
      instance.appComponent.inject(this)
   }

   override fun attachView(view: IAllCongratulationsListViewModel) {
      super.attachView(view)
      viewState?.setData()
   }
}